### ISSUE TITLE
Implement Declarative Config (YAML Training)

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#2]
Implement Ludwig-style declarative configuration system as outlined in entrenar-spec.md Phase 6 (ENT-033 to ENT-036).

## Scope

Create a YAML-based configuration system that allows users to define complete training pipelines declaratively.

## Deliverables

1. **YAML Schema & Deserialization** (ENT-033)
   - `src/config/schema.rs` - YAML schema definitions
   - TrainSpec, ModelRef, OptimSpec, DataConfig structs
   - Serde deserialization support

2. **Auto-feature Type Inference** (ENT-034)
   - Infer feature types from data (numerical, categorical, binary)
   - Support for Parquet, CSV, JSON data sources
   - Auto-detection of dtypes

3. **Config Validation** (ENT-035)
   - Type checking (lr > 0, batch_size > 0)
   - Path validation (files exist)
   - Range validation (epochs, learning rate bounds)

4. **Single-command Training** (ENT-036)
   - `train_from_yaml(path)` entry point
   - Load config → build model → train → save
   - Integration with existing autograd/optim/lora modules

## Example YAML

```yaml
model:
  path: llama-3-7b.gguf
  layers: [q_proj, k_proj, v_proj, o_proj]

data:
  train: train.parquet
  val: val.parquet
  batch_size: 8
  auto_infer_types: true

optimizer:
  name: adam
  lr: 1e-4
  beta1: 0.9
  beta2: 0.999

lora:
  rank: 64
  alpha: 16
  target_modules: [q_proj, v_proj]
```

## Quality Gates

- TDG Score ≥ 90 (A- grade)
- Test Coverage ≥ 95%
- Config validation tests (invalid configs rejected)
- End-to-end example (`examples/train_from_yaml.rs`)
- Documentation in `book/src/guide/declarative-config.md`

## References

- entrenar-spec.md lines 339-411 (Phase 6)
- Ludwig AutoML for inspiration
